web/wp-content/themes/dctx-v2/inc/hooks.php

```diff
@@ -0,0 +1,415 @@
+}
+add_action( 'wp_footer', 'dctx_display_customizer_footer_scripts', 999 );
+
+/**
+ * Adds OG tags to the head for better social sharing.
+ *
+ * @return string Just an empty string if Yoast is found.
+ * @author Corey Collins
+ */
+function dctx_add_og_tags() {
+
+	// Bail if Yoast is installed, since it will handle things.
+	if ( class_exists( 'WPSEO_Options' ) ) {
+		return '';
+	}
+
+	// Set a post global on single posts. This avoids grabbing content from the first post on an archive page.
+	if ( is_singular() ) {
+		global $post;
+	}
+
+	// Get the post content.
+	$post_content = ! empty( $post ) ? $post->post_content : '';
+
+	// Strip all tags from the post content we just grabbed.
+	$default_content = ( $post_content ) ? wp_strip_all_tags( strip_shortcodes( $post_content ) ) : $post_content;
+
+	// Set our default title.
+	$default_title = get_bloginfo( 'name' );
+
+	// Set our default URL.
+	$default_url = get_permalink();
+
+	// Set our base description.
+	$default_base_description = ( get_bloginfo( 'description' ) ) ? get_bloginfo( 'description' ) : esc_html__( 'Visit our website to learn more.', 'dctx' );
+
+	// Set the card type.
+	$default_type = 'article';
+
+	// Get our custom logo URL. We'll use this on archives and when no featured image is found.
+	$logo_id    = get_theme_mod( 'custom_logo' );
+	$logo_image = ( $logo_id ) ? wp_get_attachment_image_src( $logo_id, 'full' ) : '';
+	$logo_url   = ( $logo_id ) ? $logo_image[0] : '';
+
+	// Set our final defaults.
+	$card_title            = $default_title;
+	$card_description      = $default_base_description;
+	$card_long_description = $default_base_description;
+	$card_url              = $default_url;
+	$card_image            = $logo_url;
+	$card_type             = $default_type;
+
+	// Let's start overriding!
+	// All singles.
+	if ( is_singular() ) {
+
+		if ( has_post_thumbnail() ) {
+			$card_image = get_the_post_thumbnail_url();
+		}
+	}
+
+	// Single posts/pages that aren't the front page.
+	if ( is_singular() && ! is_front_page() ) {
+
+		$card_title            = get_the_title() . ' - ' . $default_title;
+		$card_description      = ( $default_content ) ? wp_trim_words( $default_content, 53, '...' ) : $default_base_description;
+		$card_long_description = ( $default_content ) ? wp_trim_words( $default_content, 140, '...' ) : $default_base_description;
+	}
+
+	// Categories, Tags, and Custom Taxonomies.
+	if ( is_category() || is_tag() || is_tax() ) {
+
+		$term_name      = single_term_title( '', false );
+		$card_title     = $term_name . ' - ' . $default_title;
+		$specify        = ( is_category() ) ? esc_html__( 'categorized in', 'dctx' ) : esc_html__( 'tagged with', 'dctx' );
+		$queried_object = get_queried_object();
+		$card_url       = get_term_link( $queried_object );
+		$card_type      = 'website';
+
+		// Translators: get the term name.
+		$card_long_description = $card_description = sprintf( esc_html__( 'Posts %1$s %2$s.', 'dctx' ), $specify, $term_name );
```

Comment:
:no_entry_sign: **Error**: Assignments must be the first block of code on a line (*Squiz.PHP.DisallowMultipleAssignments.Found*).